cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/AbstractClusterOperations.java

```diff
@@ -265,36 +316,36 @@ protected String nameFromLabels(R resource) {
      * @param namespace The namespace
      * @param labels The labels
      */
-    public abstract void reconcile(String namespace, Map<String, String> labels);
+    public abstract void reconcileAll(String namespace, Map<String, String> labels);
 
     /**
-     * This is provided for subclasses to help them implement the public {@link #reconcile(String, Map)}.
+     * This is provided for subclasses to help them implement the public {@link #reconcileAll(String, Map)}.
      * It obtains a list of ConfigMaps (the desired state of the cluster)
-     * and a list of other resources (of type {@link R}) (the actual state of the cluster) and determines,
-     * by comparing their names which clusters need to be created, updated or deleted.
+     * and a list of other resources (of type {@link R}) (the actual state of the cluster) and then calls,
+     * for each obtained cluster name, the corresponding {@link #reconcile(String, String, String)}
+     *
      * @param namespace The namespace
-     * @param labels The labels to use to select the ConfigMap and other resources to be compared.
-     * @param type The {@code strimzi.io/type} of the ConfigMaps and other resources
+     * @param labels The labels
+     * @param clusterType The type of the cluster
      */
-    protected void reconcile(String namespace, Map<String, String> labels, String type) {
-        log.info("Reconciling {} clusters ...", clusterDescription);
-
-        Map<String, String> kafkaLabels = new HashMap<>(labels);
-        kafkaLabels.put(ClusterController.STRIMZI_TYPE_LABEL, type);
+    protected void reconcileAll(String namespace, Map<String, String> labels, String clusterType) {
 
-        List<ConfigMap> cms = configMapOperations.list(namespace, kafkaLabels);
-        List<R> resources = getResources(namespace, kafkaLabels);
+        Map<String, String> newLabels = new HashMap<>(labels);
+        newLabels.put(ClusterController.STRIMZI_TYPE_LABEL, clusterType);
 
+        // get ConfigMap for the corresponding cluster type
+        List<ConfigMap> cms = configMapOperations.list(namespace, newLabels);
         Set<String> cmsNames = cms.stream().map(cm -> cm.getMetadata().getName()).collect(Collectors.toSet());
-        Set<String> resourceNames = resources.stream().map(cm -> cm.getMetadata().getLabels().get(ClusterController.STRIMZI_CLUSTER_LABEL)).collect(Collectors.toSet());
 
-        List<ConfigMap> addList = cms.stream().filter(cm -> !resourceNames.contains(cm.getMetadata().getName())).collect(Collectors.toList());
-        List<ConfigMap> updateList = cms.stream().filter(cm -> resourceNames.contains(cm.getMetadata().getName())).collect(Collectors.toList());
-        List<R> deletionList = resources.stream().filter(dep -> !cmsNames.contains(dep.getMetadata().getLabels().get(ClusterController.STRIMZI_CLUSTER_LABEL))).collect(Collectors.toList());
+        // get StatefulSets for the corresponding cluster name (they are part of)
```

Comment:
Comment is wrong, it's not necessarily StatefulSets it's Rs we're getting here.

Reply:
Right sorry !